src/Compilers/Core/MSBuildTaskTests/TestUtilities/ErrorLoggerEngine.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Reflection;
+using System.Text;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests
+{
+    /// <summary>
+    /// An engine to output event messages as MSBuild does to test Vbc.ParseVBErrorOrWarning.
+    /// </summary>
+    internal sealed class ErrorLoggingEngine : IBuildEngine
+    {
+        private StringBuilder _log = new StringBuilder();
+        public MessageImportance MinimumMessageImportance = MessageImportance.Low;
+        private readonly MethodInfo _formatErrorMethod;
+        private readonly MethodInfo _formatWarningMethod;
+
+        public ErrorLoggingEngine()
+        {
+            // Use the formatting from Microsoft.Build.Shared.EventArgsFormatting.
+            var assembly = Assembly.LoadFrom("Microsoft.Build.dll");
+            var formattingClass = assembly.GetType("Microsoft.Build.Shared.EventArgsFormatting") ?? throw new Exception("Could not find EventArgsFormatting type");
+            _formatErrorMethod = formattingClass.GetMethod("FormatEventMessage", BindingFlags.Static | BindingFlags.NonPublic, null, CallingConventions.Any,
+                new Type[] { typeof(BuildErrorEventArgs) }, null) ?? throw new Exception("Could not find FormatEventMessage(BuildErrorEventArgs).");
+            _formatWarningMethod = formattingClass.GetMethod("FormatEventMessage", BindingFlags.Static | BindingFlags.NonPublic, null, CallingConventions.Any,
+                new Type[] { typeof(BuildWarningEventArgs) }, null) ?? throw new Exception("Could not find FormatEventMessage(BuildWarningEventArgs).");
```

Comment:
Using reflection here concerns me greatly. @rainersigwald, do you have suggestions for a better way to accomplish this?

Reply:
For context, we're trying to test our parsing of msbuild error message output in Vbc.exe

Reply:
Sorry, missed this. I agree that this feels icky but I'm afraid I don't have a better idea at the moment.